Builds/CMake/deps/Secp256k1.cmake

```diff
@@ -2,34 +2,58 @@
    NIH dep: secp256k1
 #]===================================================================]
 
-add_library (secp256k1 STATIC
-  src/secp256k1/src/secp256k1.c)
-target_compile_definitions (secp256k1
-  PRIVATE
-    USE_NUM_NONE
-    USE_FIELD_10X26
-    USE_FIELD_INV_BUILTIN
-    USE_SCALAR_8X32
-    USE_SCALAR_INV_BUILTIN)
-target_include_directories (secp256k1
-  PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
-    $<INSTALL_INTERFACE:include>
-  PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/src/secp256k1)
-target_compile_options (secp256k1
-  PRIVATE
-    $<$<BOOL:${MSVC}>:-wd4319>
-    $<$<NOT:$<BOOL:${MSVC}>>:
-      -Wno-deprecated-declarations
-      -Wno-unused-function
-    >
-    $<$<BOOL:${is_gcc}>:-Wno-nonnull-compare>)
-add_library (NIH::secp256k1 ALIAS secp256k1)
-target_link_libraries (ripple_libs INTERFACE NIH::secp256k1)
+add_library (secp256k1_lib STATIC IMPORTED GLOBAL)
+
+if (NOT WIN32)
+  find_package(secp256k1)
+endif()
+
+
+if(secp256k1)
+  set_target_properties (secp256k1_lib PROPERTIES
+    IMPORTED_LOCATION_DEBUG
+      ${secp256k1}
+    IMPORTED_LOCATION_RELEASE
+      ${secp256k1}
+    INTERFACE_INCLUDE_DIRECTORIES
+      ${SECP256K1_INCLUDE_DIR})
+
+  add_library (secp256k1 ALIAS secp256k1_lib)
+  add_library (NIH::secp256k1 ALIAS secp256k1_lib)
+
+else()
+  set(INSTALL_SECP256K1 true)
+
+  add_library (secp256k1 STATIC
+    src/secp256k1/src/secp256k1.c)
+  target_compile_definitions (secp256k1
+    PRIVATE
+      USE_NUM_NONE
```

Comment:
For libraries like this that have a number of preprocessor options/variants, I wonder how the installed library might differ. I suspect most of these are just speed optimizations, but it brings up the question of how the prebuilt/installed library could differ functionally from the one we build. Same situation for sqlite where there is the `SQLITE_THREADSAFE` option. I assume it impacts the binary output (not just headers) and it's not obvious to me which option a prebuilt library might choose for that setting.  I'm not too worried about all this, but it is something to keep in mind if we think any of those build options matter significantly to our application.

Reply:
@mellery451 thanks for the review and agree. Many Linux distros include default flags for enhanced security, platform comparability, etc. Furthermore builds for different architectures (x86, ARM, etc) will often entail different compiler flags.

Unfortunately there is no way to know the impact until we actually try to compile (and use) the binaries on those platforms.